backend/app/services/memory/manager.py

```
@@ -0,0 +1,403 @@
+                # If we deleted fewer than batch_size memories, we've reached the end
+                if len(memories) < batch_size:
+                    break
+
+            logger.info(
+                "Cleaned up %d memories for task %s (%d errors)",
+                total_delete_count,
+                task_id,
+                total_error_count,
+            )
+
+            return total_delete_count
```

Ruff (0.14.11)

Consider moving this statement to an `else` block

(TRY300)